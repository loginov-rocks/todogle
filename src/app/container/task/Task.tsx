import { AppBar, Toolbar, Typography } from '@material-ui/core';
import * as React from 'react';

interface Props {
  drawerToggle?: JSX.Element;
}

const Task = ({ drawerToggle }: Props) => (
  <>

    <AppBar position="static">
      <Toolbar>
        {drawerToggle}
        <Typography component="h1" variant="h6">
          Task
        </Typography>
      </Toolbar>
    </AppBar>

    <main>
      Task
    </main>

  </>
);

export default Task;
