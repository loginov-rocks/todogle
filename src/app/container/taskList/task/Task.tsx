import { Button } from '@material-ui/core';
import * as React from 'react';
import { useHistory } from 'react-router-dom';

import * as R from '../../../../routes';
import gapi from '../../../../services/gapi';
import { TaskListResource } from '../../../../services/gapi/TaskListResource';
import { TaskResource } from '../../../../services/gapi/TaskResource';

interface Props {
  onDelete: (id: string) => void;
  task: TaskResource;
  taskList: TaskListResource;
}

const Task = ({ onDelete, task, taskList }: Props) => {
  const history = useHistory();

  const handleClick = () => {
    history.push(R.toTask(taskList.id, task.id));
  };

  const handleDelete = () => {
    gapi.deleteTask(taskList.id, task.id)
      .then(() => onDelete(task.id));
  };

  return (
    <div>
      <i onClick={handleClick}>{task.title}</i>
      <Button onClick={handleDelete} variant="contained">Delete</Button>
    </div>
  );
};

export default Task;
