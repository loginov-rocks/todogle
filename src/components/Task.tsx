import * as React from 'react';

import gapi from '../services/gapi';
import { TaskListResource } from '../services/gapi/TaskListResource';
import { TaskResource } from '../services/gapi/TaskResource';

interface Props {
  onDelete: (id: string) => void;
  task: TaskResource;
  taskList: TaskListResource;
}

export default class Task extends React.Component<Props> {
  constructor(props: Props) {
    super(props);

    this.handleDelete = this.handleDelete.bind(this);
  }

  handleDelete() {
    const { onDelete, task, taskList } = this.props;

    gapi.deleteTask(taskList.id, task.id)
      .then(() => onDelete(task.id));
  }

  render() {
    const { task } = this.props;

    return (
      <div>
        <i>{task.title}</i>
        <button onClick={this.handleDelete}>Delete</button>
      </div>
    );
  }
}
