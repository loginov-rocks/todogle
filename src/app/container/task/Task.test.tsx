import { shallow } from 'enzyme';
import * as React from 'react';

import Task from './Task';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Task />,
  );

  expect(wrapper).toMatchSnapshot();
});
