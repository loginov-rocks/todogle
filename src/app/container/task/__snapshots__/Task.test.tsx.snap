// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        component="h1"
        variant="h6"
      >
        Task
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main>
    Task
  </main>
</Fragment>
`;
